pages/allCards.js

```diff
@@ -1,24 +1,39 @@
 import CardList from "../components/CardList/CardList";
 import Navigation from "../components/Navigation/Navigation";
 import { useSession } from "next-auth/react";
+import SearchBar from "@/components/SearchBar/SearchBar";
+import useSWR from "swr";
+import { useState } from "react";
+
+const fetcher = (url) => fetch(url).then((response) => response.json());
 
 export default function AllCardsPage({
-  cards,
   deleteCard,
   onToggle,
   resetCard,
   toggleCardDirection,
 }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const { data: cards, isLoading: isLoadingCards } = useSWR(
+    searchTerm ? "/api/cards?filter=" + searchTerm : "/api/cards",
+    fetcher
+  );
+  const handleSearch = (event) => {
+    setSearchTerm(event.target.value);
+  };
+
   return (
     <>
```

Comment:
I think you don't need the fragment here because the function returns only one element.